core/src/test/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSenderFactoryTest.java

```diff
@@ -0,0 +1,41 @@
+package com.microsoft.applicationinsights.internal.channel.common;
+
+import com.microsoft.applicationinsights.internal.reflect.ClassDataUtils;
+import com.microsoft.applicationinsights.internal.reflect.ClassDataVerifier;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.lang.reflect.Field;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.anyString;
+
+public final class ApacheSenderFactoryTest {
+    @Test
+    public void testBadJar() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
I'm not sure that I'd name these tests Bad and Good jars as we support both versions so both should be considered as good...
Instead I would rename these tests to `testV42Lib` or `testNewJar` (less preferred for the case one day we'll decide to support even older/newer version of this library.